src/pile.sol

```diff
@@ -190,18 +200,22 @@ contract Pile is DSNote {
         return int(Balance) - int(tkn.balanceOf(address(this))); // safemath
     }
 
+    function initLoan(uint loan,uint wad, uint chi) internal {
+        loans[loan].chi = chi;
+        loans[loan].debt = add(loans[loan].debt, wad);
+        loans[loan].balance = add(loans[loan].balance, wad);
+        Balance = add(Balance, wad);
+
+    }
+
     // borrow() creates a debt by the borrower for the specified amount. 
     function borrow(uint loan, uint wad) public auth note {
         uint fee = loans[loan].fee;
         drip(fee);
-        loans[loan].chi = fees[fee].chi;
-        collect(loan);
```

Comment:
the `collect` call was unnecessary.

`collect` updates `loans[loan].debt` `loans[loan].chi`.

`loans[loan].chi` is already assigned in the new `initLoan` method previously in line 197.

`loans[loan].debt` always stays zero. Therefore the collect call is not needed.